R/utils.R

```diff
@@ -216,6 +216,7 @@ prepare_precompiled <- function(cpp_options = list(), quiet = FALSE) {
     model_header_gch_used <- FALSE
   }
   if (!file.exists(main_path_w_flags)) {
+    message("Re-compiling the main object file and precompiled headers. This might take up to a few minutes ...")
```

Comment:
I think it could be good to explain a bit more in the message, for example:

```r
message(
"Compiling the main object file and precompiled headers (may take up to a few minutes). ",
"This is only necessary the first time a model is compiled after installation or when certain ",
"C++ options are changed."
)
```

It's a bit long but I think more clear. What do you think? 

Also, do you know for which compiler option changes this will be necessary?

Reply:
This looks good. Will update in the morning.
This is be required for threads, mpi and opencl.

Reply:
What about: 
```R
message(
      "Compiling the main object file and precompiled headers (may take up to a few minutes). ",
      "This is only necessary for the first compilation after installation or when ",
      "threading, MPI or OpenCL are used for the first time."
    )
```